### PR TITLE
Weather save 

### PR DESCRIPTION
**Pull request by yuriyKolomiyets**

### CODE REVIEW

File: src/main/java/com/example/travelservice/services/TripServiceImpl.java [hunk 1]
```diff
@@ -51,4 +48,18 @@ public Trip updateTrip(Long id, Trip trip) {
             throw new NotFoundException("Trip Not Found. For ID value: " + id.toString());
         }
     }
+
+    @Override
+    public Long findId(Trip source) {
+        Trip trip = tripRepository.findByStartDateAndLocation(source.getStartDate(), source.getLocation()).get(0);
```
Comment (bizianov) [inline src/main/java/com/example/travelservice/services/TripServiceImpl.java]: refactor as discussed on the call: use id

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java [hunk 2]
```diff
@@ -2,24 +2,32 @@
...
-@RequiredArgsConstructor
 @Slf4j
 public class WeatherServiceImpl implements WeatherService {
 
     private final WeatherFromApiService weatherFromApiService;
     private final WeatherChannels weatherChannels;
+    private final TripService tripService;
+
+    public WeatherServiceImpl(WeatherFromApiService weatherFromApiService, WeatherChannels weatherChannels,
+                              @Lazy TripService tripService) {
```
Comment (bizianov) [inline src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: refactor to use ApplicationEventPublisher

**Review by bizianov — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/travelservice/services/TripServiceImpl.java (hunk 1 above)
Comment (yuriyKolomiyets) [reply src/main/java/com/example/travelservice/services/TripServiceImpl.java]: done

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java (hunk 2 above)
Comment (yuriyKolomiyets) [reply src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: done

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java [hunk 3]
```diff
@@ -2,24 +2,34 @@
...
 
 @Service
-@RequiredArgsConstructor
 @Slf4j
 public class WeatherServiceImpl implements WeatherService {
 
     private final WeatherFromApiService weatherFromApiService;
     private final WeatherChannels weatherChannels;
+    @Autowired
+    private ApplicationEventPublisher applicationEventPublisher;
```
Comment (bizianov) [inline src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: make it final and inject via constructor as other beans here

File: src/main/java/com/example/travelservice/dtoconverters/TripConverter.java [hunk 4]
```diff
@@ -36,14 +32,15 @@ public Trip convertTripDtoToTrip (TripDto source) {
                 Enum.valueOf(Meal.class, source.getMeal()));
     }
 
-    public WeatherRequest convertTripToWeatherRequest (Trip source) {
+    public WeatherRequest convertTripToWeatherRequest (Trip source, Long tripId) {
```
Comment (bizianov) [inline src/main/java/com/example/travelservice/dtoconverters/TripConverter.java]: why you need tripId here if you have trip itself?

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java [hunk 5]
```diff
@@ -28,8 +38,15 @@ public List<WeatherDto> getWeatherThroughRest(Trip trip) {
...
     @StreamListener(WeatherChannels.WEATHER_RESPONSE_INPUT_CHANNEL)
-    public void listenWeatherResponse(List<WeatherDto> weathers) {
-        log.info("Got response for weatherApi {}", weathers);
+    public void listenWeatherResponse(WeatherResponse weatherResponses) {
+        log.info("Got response for weatherApi {}", weatherResponses);
+        publishCustomEvent(weatherResponses.getTripId(), weatherResponses.getWeatherDtoList());
+    }
+
+    public void publishCustomEvent(Long tripId, List<WeatherDto> weatherDtoList) {
+        System.out.println("Publishing custom event. ");
```
Comment (bizianov) [inline src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: use logger instead of sout

**Review by bizianov — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java (hunk 3 above)
Comment (yuriyKolomiyets) [reply src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: make sense

File: src/main/java/com/example/travelservice/dtoconverters/TripConverter.java (hunk 4 above)
Comment (yuriyKolomiyets) [reply src/main/java/com/example/travelservice/dtoconverters/TripConverter.java]: resolved

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java (hunk 3 above)
Comment (yuriyKolomiyets) [reply src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: resolved

File: src/main/java/com/example/travelservice/services/WeatherServiceImpl.java (hunk 5 above)
Comment (yuriyKolomiyets) [reply src/main/java/com/example/travelservice/services/WeatherServiceImpl.java]: log.info()

**Review by bizianov — APPROVED** [review]

**Pull request merged**